core/src/dird/ua_select.cc

```diff
@@ -1107,6 +1107,58 @@ void AddPrompt(UaContext* ua, std::string&& prompt)
   AddPrompt(ua, p.c_str());
 }
 
+
+/**
+ * Formats the prompts of a UaContext to be displayed in a multicolumn output
+ * when possible
+ */
+std::string FormatMulticolumnPrompts(const UaContext* ua,
+                                     const int windowsize,
+                                     const int min_lines_threshold)
+{
+  size_t max_prompt_length = 1;
+
+  const int max_prompt_index_length = std::to_string(ua->num_prompts).length();
+
+  for (int i = 1; i < ua->num_prompts; i++) {
+    if (strlen(ua->prompt[i]) > max_prompt_length) {
+      max_prompt_length = strlen(ua->prompt[i]);
+    }
+  }
+
+  const int extra_formatting_characters = 4;
+  const int max_formatted_prompt_length = max_prompt_length
+                                          + max_prompt_index_length
+                                          + extra_formatting_characters;
+  const int prompts_perline
+      = std::max(1, windowsize / (max_formatted_prompt_length - 1));
+
+  char formatted_prompt[max_formatted_prompt_length];
```

Comment:
what @franku meant was that you could use `std::vector<char> formatted_prompt(max_formatted_prompt_length)` instead of a `char[]`.
Then you would use `formatted_prompt.data()` and `formatted_prompt.size()` as parameters to `snprintf()` below.
This may be advantageous because a `std::vector<char>` will never decay to a pointer.

For your specific use I think using `char[]` is allright, however it is worth keeping in mind that you can always use a `std::vector<T>` instead of a `T[]` to get a safe buffer that always knows its size and can be resized easily.